### PR TITLE
`ServerDriveQueryVolumeInformationRequest` and `ClientDriveQueryVolumeInformationResponse`

### PR DESCRIPTION
**Pull request by ibeckermayer**
Corresponds with https://github.com/Devolutions/IronRDP/pull/296

### CODE REVIEW

File: lib/srv/desktop/rdp/rdpclient/src/client.rs [hunk 1]
```diff
@@ -942,6 +942,7 @@ fn create_config(width: u16, height: u16, pin: String) -> Config {
         platform: MajorPlatformType::UNSPECIFIED,
```
Comment (zmb3) [inline lib/srv/desktop/rdp/rdpclient/src/client.rs:942]: What is this `client_dir` DLL?

File: lib/srv/desktop/rdp/rdpclient/src/client.rs [hunk 2]
```diff
@@ -942,6 +942,7 @@ fn create_config(width: u16, height: u16, pin: String) -> Config {
...
         no_server_pointer: false,
         autologon: true,
+        pointer_software_rendering: false,
```
Comment (zmb3) [inline lib/srv/desktop/rdp/rdpclient/src/client.rs:945]: Unrelated change?

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs [hunk 3]
```diff
@@ -719,7 +825,7 @@ impl FilesystemBackend {
         file: &FileCacheObject,
     ) -> PduResult<()> {
         let file_fso_size: i64 = cast_length!(
-            "FilesystemBackend::make_client_drive_query_information_response",
+            "FilesystemBackend::send_file_standard_info",
```
Comment (zmb3) [inline lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:828]: I wonder if something like https://stackoverflow.com/a/40234666/914991 would be worthwhile for a lot of these areas where you're manually writing the name of the enclosing function. These are prone to copy-pasta errors.

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs [hunk 4]
```diff
@@ -352,6 +355,109 @@ impl FilesystemBackend {
...
+                            volume_creation_time: cast_length!(
+                                "FilesystemBackend::handle_query_volume_req",
+                                "dir.fso.last_modified",
+                                dir.fso.last_modified
+                            )?,
+                            // Not sure why the `& 0xffff` is necessary, just copying FreeRDP
+                            // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_main.c#L492
+                            // u32::MAX is given due to the fact that FreeRDP uses it as a fallback:
+                            // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/winpr/libwinpr/file/file.c#L1018-L1021
+                            volume_serial_number: u32::MAX & 0xffff,
```
Comment (probakowski) [inline lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:376]: From http://msdn.microsoft.com/en-us/library/cc232108.aspx:
> VolumeSerialNumber (4 bytes): A 32-bit unsigned integer that contains the serial number of the volume. The serial number is an opaque value generated by the file system at format time, and is not necessarily related to any hardware serial number for the device on which the file system is located. No specific format or content of this field is required for protocol interoperation. This value is not required to be unique.

So you can use any value you want, `0xfff` is as good as any but `u32::MAX &` is redundant

File: lib/srv/desktop/rdp/rdpclient/src/client.rs (hunk 2 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/client.rs:947]: This (as well as changes in `web/packages/teleport/src/ironrdp/src/lib.rs`) was necessary to keep up with a changing IronRDP api. Ideally I would have done them as a separate commit, though it looks like I just lumped them all together.

I took a closer look at these settings and updated with 2e911c069f67c1aa7e07193db6809f3fb1b585bb

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs (hunk 3 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:828]: Good call, I'll add that to a PR on the top of the stack.

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs (hunk 4 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:376]: 11f4d6aff990b0da0cf420d50bff76d45b7f39f7

Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs]: 11f4d6aff990b0da0cf420d50bff76d45b7f39f7

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs [hunk 5]
```diff
@@ -352,6 +355,109 @@ impl FilesystemBackend {
...
+                    .into(),
+                ),
+                _ => None,
+            };
+
+            let io_status = if buffer.is_some() {
+                NtStatus::SUCCESS
+            } else {
+                NtStatus::UNSUCCESSFUL
+            };
```
[the remark this answers is not in the document]
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs]: Fixed in a larger refactor e79a54f4c6d44d5defa1cfbfc53678b058dd3eea

**Review by probakowski — APPROVED** [review]
